packages/flutter_test/lib/src/binding.dart

```diff
@@ -1095,28 +1095,45 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
 
     addTime(additionalTime);
 
-    return realAsyncZone.run<Future<T?>>(() async {
+    return realAsyncZone.run<Future<T?>>(() {
+      final Completer<T?> result = Completer<T?>();
       _pendingAsyncTasks = Completer<void>();
-      T? result;
       try {
-        result = await callback();
+        callback().then((T? value) {
+          result.complete(value);
+        }).catchError((Object exception, StackTrace stack) {
```

Comment:
Technically, there's an unnecessary closure here. I think, this could just be 

```dart
callback().then(result.complete).catchError(...)
```

Reply:
done